src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1122,38 +1125,174 @@ class spell_pri_power_word_shield : public SpellScriptLoader
+    }
+
+protected:
+    SpellInfo const* _spellInfoHeal;
+    SpellEffectInfo const* _healEffectDummy;
+};
+
+// 33076 - Prayer of Mending
+class spell_pri_prayer_of_mending : public SpellScriptLoader
+{
+public:
+    spell_pri_prayer_of_mending() : SpellScriptLoader("spell_pri_prayer_of_mending") { }
+
+    class spell_pri_prayer_of_mending_SpellScript : public spell_pri_prayer_of_mending_SpellScriptBase
+    {
+        PrepareSpellScript(spell_pri_prayer_of_mending_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_AURA });
+        }
+
+        void HandleEffectDummy(SpellEffIndex effIndex)
+        {
+            SpellEffectInfo const* effectInfo = GetEffectInfo(effIndex);
+            if (!effectInfo)
+                return;
+
+            Unit* target = GetExplTargetUnit();
+            if (!target)
+                return;
+
+            Unit* caster = GetCaster();
+            assert(caster != null);
+
+            CastPrayerOfMendingAura(caster, target, effectInfo->BasePoints);
+        }
+
+        void Register() override
+        {
+            OnEffectHit += SpellEffectFn(spell_pri_prayer_of_mending_SpellScript::HandleEffectDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_pri_prayer_of_mending_SpellScript();
+    }
+};
+
+// 41635 - Prayer of Mending (Aura) - SPELL_PRIEST_PRAYER_OF_MENDING_AURA
+class spell_pri_prayer_of_mending_aura : public SpellScriptLoader
 {
     public:
-        spell_pri_prayer_of_mending_heal() : SpellScriptLoader("spell_pri_prayer_of_mending_heal") { }
+        spell_pri_prayer_of_mending_aura() : SpellScriptLoader("spell_pri_prayer_of_mending_aura") { }
 
-        class spell_pri_prayer_of_mending_heal_SpellScript : public SpellScript
+        class spell_pri_prayer_of_mending_aura_AuraScript : public AuraScript
         {
-            PrepareSpellScript(spell_pri_prayer_of_mending_heal_SpellScript);
+            PrepareAuraScript(spell_pri_prayer_of_mending_aura_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_HEAL });
+            }
 
-            void HandleHeal(SpellEffIndex /*effIndex*/)
+            void HandleHeal(ProcEventInfo & /*eventInfo*/)
             {
-                if (Unit* caster = GetOriginalCaster())
-                {
-                    if (AuraEffect* aurEff = caster->GetAuraEffect(SPELL_PRIEST_T9_HEALING_2P, EFFECT_0))
+                // Caster: the player(priest) that casters the prayer of mending
+                // Owner: the player that currently has the prayer of mending aura on him
+                Unit* owner = GetUnitOwner();
+                assert(owner != null);
```

Comment:
this will never happen, assert not required

Reply:
He should use GetTarget() instead which will return the unit that has the aura.

Reply:
@Ovahlord What's the difference? the comments on the properties sounds vaguely similar, and using GetUnitOwner() seems to use the correct unit from the tests I've made.